lib/rides/models/ride.dart

```diff
@@ -127,6 +127,23 @@ class Ride extends Trip {
     return Ride.fromJsonList(await supabaseClient.from('rides').select().eq('rider_id', userId));
   }
 
+  @override
+  bool isDifferentFrom(Trip other) {
+    if (other is! Ride) return true;
+    Ride ride = other;
+    return start != ride.start ||
```

Comment:
you can also call super.isDifferentFrom (equals) for the parameters that are in trip